src/routes/Scanner.tsx

```diff
@@ -1,9 +1,48 @@
 import Reader from "~/components/Reader";
-import { createEffect, createSignal, Show } from "solid-js";
+import { createEffect, createSignal, onMount, Show } from "solid-js";
 import { useNavigate } from "solid-start";
 import { Button } from "~/components/layout";
+import init, { PaymentParams } from "@mutinywallet/waila-wasm";
+import { showToast } from "~/components/Toaster";
+import { useMegaStore } from "~/state/megaStore";
+import { Result } from "~/utils/typescript";
+
+export type ParsedParams = {
+    address?: string;
+    invoice?: string;
+    amount_sats?: bigint;
+    network?: string;
+    memo?: string;
+}
+
+export function toParsedParams(str: string, ourNetwork: string): Result<ParsedParams> {
+    const params = new PaymentParams(str || "")
+
+    console.log("params network:", params.network)
+    console.log("our network:", ourNetwork)
+
+    // TODO: "testnet" and "signet" are encoded the same I guess?
+    if (params.network === "testnet" || params.network === "signet") {
+        if (ourNetwork === "signet") {
+            // noop
+        }
+    } else if (params.network !== ourNetwork) {
+        return { ok: false, error: new Error(`Destination is for ${params.network} but you're on ${ourNetwork}`) }
+    }
```

Comment:
We do these checks in the backend, probably okay for you to skip

Reply:
If i do it here i fails in the scanner ui instead of pretending like it's valid until they send, is there a way i can ask the backend for help?

Reply:
Okay that makes sense. Maybe keep this for now and I can add something to waila for this